pkg/storage/watch_cache_test.go

```diff
@@ -230,6 +231,25 @@ func TestEvents(t *testing.T) {
 	}
 }
 
+func TestWaitUntilFreshAndList(t *testing.T) {
+	store := newWatchCache(3)
+
+	// In background, wait some time and then update the store.
```

Comment:
Not that I can think of. I don't particularly think this is worth having since this is a fairly trivial use of condition variables.